src/Categories/Functor/Profunctor.agda

```diff
@@ -3,14 +3,399 @@
+    ; id = λ {c} → record
+      { twiner = Category.id C
+      ; in-tertwines = let x = F₀ (T.rendezvous c) in sym x (identity (refl x))
+      ; out-ertwines = let x = G.₀ (T.rendezvous c) in sym x (G.identity (refl x)) }
+    ; _∘_ = λ {a b c} f g → record
+      { twiner = twiner f ∘ twiner g
+      ; in-tertwines = let open SetoidR (F₀ (T.rendezvous c)) in
+        begin
+          T.in-side c
+        ≈⟨ in-tertwines f ⟩
+          F₁ (twiner f) ⟨$⟩ T.in-side b
+        ≈⟨ F-resp-≈ Equiv.refl (in-tertwines g) ⟩
+          F₁ (twiner f) ⟨$⟩ (F₁ (twiner g) ⟨$⟩ T.in-side a)
+        ≈⟨ sym (F₀ (T.rendezvous c)) (homomorphism (refl (F₀ (T.rendezvous a)))) ⟩
+          F₁ (twiner f ∘ twiner g) ⟨$⟩ T.in-side a
+        ∎
+      ; out-ertwines = let open SetoidR (G.₀ (T.rendezvous a)) in
+        begin
+          T.out-side a
+        ≈⟨ out-ertwines g ⟩
+          G.₁ (twiner g) ⟨$⟩ T.out-side b
+        ≈⟨ G.F-resp-≈ Equiv.refl (out-ertwines f) ⟩
+          G.₁ (twiner g) ⟨$⟩ (G.₁ (twiner f) ⟨$⟩ T.out-side c)
+        ≈⟨ sym (G.₀ (T.rendezvous a)) (G.homomorphism (refl (G.₀ (T.rendezvous c)))) ⟩
+          G.₁ (twiner f ∘ twiner g) ⟨$⟩ T.out-side c
+        ∎
+      }
+    ; assoc = assoc
+    ; sym-assoc = sym-assoc
+    ; identityˡ = identityˡ
+    ; identityʳ = identityʳ
+    ; identity² = identity²
+    ; equiv = record { refl = Equiv.refl ; sym = Equiv.sym ; trans = Equiv.trans }
+    ; ∘-resp-≈ = ∘-resp-≈
+    }
+
+  setoid = ST.setoid Twines (Category.id category)
+
+-- this is the left adjoint to Disc : Setoids ⇒ Cats
+Π₀ : ∀ {o ℓ e} → Functor (Cats o ℓ e) (Setoids o (o ⊔ ℓ))
+Π₀ = record
+  { F₀ = λ C → ST.setoid (Category._⇒_ C) (Category.id C)
+  ; F₁ = λ F → record
+    { _⟨$⟩_ = Functor.F₀ F
+    ; cong = ST.gmap (Functor.F₀ F) (Functor.F₁ F)
+    }
+  ; identity = λ x → x
+  ; homomorphism = λ {_ _ _ F G} → ST.gmap (Functor.F₀ G ∙ Functor.F₀ F) (Functor.F₁ G ∙ Functor.F₁ F)
+  ; F-resp-≈ = my-resp _ _
+  }
+  where
+  my-resp : ∀ {A B : Category _ _ _} (f g : Functor A B) (f≅g : NaturalIsomorphism f g) {x y} → Plus⇔ (Category._⇒_ A) x y → Plus⇔ (Category._⇒_ B) (Functor.F₀ f x) (Functor.F₀ g y)
+  my-resp {A} {B} f g f≅g {x} {y} (Plus⇔.forth m) = Plus⇔.forth (B [ NaturalIsomorphism.⇒.η f≅g _ ∘ Functor.F₁ f m ])
+  my-resp {A} {B} f g f≅g {x} {y} (Plus⇔.back m) = Plus⇔.back (B [ Functor.F₁ f m ∘ NaturalIsomorphism.⇐.η f≅g _ ])
+  my-resp {A} {B} f g f≅g {x} {y} (Plus⇔.forth⁺ m ms) = Plus⇔.forth⁺ (B [ NaturalIsomorphism.⇒.η f≅g _ ∘ Functor.F₁ f m ]) (ST.gmap (Functor.F₀ g) (Functor.F₁ g) ms)
+  my-resp {A} {B} f g f≅g {x} {y} (Plus⇔.back⁺ m ms) = Plus⇔.back⁺ (B [ Functor.F₁ f m ∘ NaturalIsomorphism.⇐.η f≅g _ ]) (ST.gmap (Functor.F₀ g) (Functor.F₁ g) ms)
+
+_ⓟ′_ : ∀ {oC ℓC eC oD ℓD eD oE ℓE eE ℓP eP ℓQ eQ} {C : Category oC ℓC eC} {D : Category oD ℓD eD} {E : Category oE ℓE eE} (P : Profunctor ℓP eP D E) (Q : Profunctor ℓQ eQ C D) → Bifunctor (Category.op E) C (Cats (oD ⊔ ℓC ⊔ ℓD ⊔ ℓE ⊔ ℓP ⊔ ℓQ) (ℓD ⊔ eC ⊔ eD ⊔ eE ⊔ eP ⊔ eQ) (eD))
```

Comment:
I wonder if this strange form of profunctor composition is useful for anything other than building the normal one.